### PR TITLE
Add initial implementation

### PR DESCRIPTION
**Pull request by apfelbox**

### CODE REVIEW

File: src/DefaultScriptLoader.php [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    public function __construct (array $config, int $acceptedLevel)
+    {
+        for ($i = PrivacyManager::LEVEL_NECESSARY; $i <= $acceptedLevel; $i++)
+        {
+            if (!isset($config["scripts"][$i]))
+            {
+                continue;
+            }
+
+            foreach ($config["scripts"][$i] as $scriptConfig)
```
Comment (keichinger) [inline src/DefaultScriptLoader.php]: What about writing something like this:

```php
foreach ($config["scripts"][$i] ?? [] as $scriptConfig)
```

Then we can omit line 27 to 30.

**Comment by keichinger** [comment]
Other than that it looks good to me :) Like a 1:1 copy of the JavaScript-side.

From my side we're ready to merge.

File: src/DefaultScriptLoader.php (hunk 1 above)
Comment (apfelbox) [reply src/DefaultScriptLoader.php]: Because I don't think that the lost readability is helping us here.

Comment (keichinger) [reply src/DefaultScriptLoader.php]: Fair enough.

**Pull request merged**